<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="callback (Calibrated Clustering via Knockoffs) is a method for protecting
 against over-clustering by controlling for the impact of double-dipping.
 This package reproduces the figures found in the callback manuscript.">
<title>Reproducibility of callback manuscript â€¢ callbackreproducibility</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Reproducibility of callback manuscript">
<meta property="og:description" content="callback (Calibrated Clustering via Knockoffs) is a method for protecting
 against over-clustering by controlling for the impact of double-dipping.
 This package reproduces the figures found in the callback manuscript.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">callbackreproducibility</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/1_figure1.html">1. Clustering Simulated Data (Figure 1)</a>
    <a class="dropdown-item" href="articles/2_cluster-tabula-muris.html">2. Clustering Tabula Muris Tissues</a>
    <a class="dropdown-item" href="articles/3_clustering_metrics.html">3. Calculate Clustering Metrics For Tabula Muris</a>
    <a class="dropdown-item" href="articles/4_figure2.html">4. Tabula Muris Cluster Benchmarking (Figure 2 and S2-S5)</a>
    <a class="dropdown-item" href="articles/5_figure3.html">5. Avoiding Over-clustering Leads to Improved Hypothesis Generation for Downstream Analyses (Figure 3)</a>
    <a class="dropdown-item" href="articles/6_tissue_umaps.html">6. Plotting Tabula Muris UMAPs (Supplemental Figures S6-S25)</a>
    <a class="dropdown-item" href="articles/7_multi-dataset-timing.html">7. Benchmarking Runtime and Peak Memory Usage on Multiple Datasets</a>
    <a class="dropdown-item" href="articles/8_seurat_data_and_liver_timing.html">8. Plotting Runtime and Peak Memory Usage for Multiple Tissues (Supplemental Figures S26 and S27)</a>
    <a class="dropdown-item" href="articles/9_pbmc-timing.html">9. Benchmarking Runtime and Peak Memory Usage on PBMC Subsets</a>
    <a class="dropdown-item" href="articles/9_plot_pbmc_timing.html">10. Plotting Runtime and Peak Memory Usage on PBMC Subsets (Supplemental Figures S28 and S29)</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="callback-calibrated-clustering-via-knockoffs-reproducibility-">callback (Calibrated Clustering via Knockoffs) Reproducibility <img src="reference/figures/callback_logo.png" align="right" alt="" width="120"><a class="anchor" aria-label="anchor" href="#callback-calibrated-clustering-via-knockoffs-reproducibility-"></a>
</h1></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>We hope to make it as simple as possible to reproduce the results found in <code>A knockoff calibration method to avoid over-clustering in single-cell RNA-sequencing</code>. To that end, we have organized our analysis scripts as a series of <code>R</code> vignettes in this repository.</p>
</div>
<div class="section level2">
<h2 id="reproducing-figures-from-the-callback-manuscript">Reproducing Figures from the <code>callback</code> manuscript<a class="anchor" aria-label="anchor" href="#reproducing-figures-from-the-callback-manuscript"></a>
</h2>
<p>Clone the repository:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">git</span> clone https://github.com/lcrawlab/callbackreproducibility</span></code></pre></div>
<p>Then, navigate to the repo directory and launch R</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="bu">cd</span> callbackreproducibility</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">R</span></span></code></pre></div>
<p>You can build the entire website using the following line of <code>R</code> code.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">pkgdown</span><span class="fu">::</span><span class="fu"><a href="https://pkgdown.r-lib.org/reference/build_site.html" class="external-link">build_site</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Note that each <code>Rmarkdown</code> file is not fully run by <code>R</code> by default. To properly run an an <code>Rmarkdown</code> file run during the website building process, you need to remove</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>eval <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>from the file header. We do not recommend doing this for the vignettes that actually use <code>callback</code>, <code>sc-SHC</code>, and <code>CHOIR</code> for clustering because they have a long runtime. Rather, the R portions of these files should be put in a script and run using <code>Rscript</code>.</p>
</div>
<div class="section level2">
<h2 id="relevant-citations">Relevant Citations<a class="anchor" aria-label="anchor" href="#relevant-citations"></a>
</h2>
<p><code>callback</code> is currently on the bioRxiv, <a href="%5Bhttps://duckduckgo.com%5D(https://www.biorxiv.org/content/10.1101/2024.03.08.584180v1)" class="external-link">here</a>.</p>
<p>A. DenAdel, M. Ramseier, A. Navia, A. Shalek, S. Raghavan, P. Winter, A. Amini, and L. Crawford. A knockoff calibration method to avoid over-clustering in single-cell RNA-sequencing. <em>bioRxiv</em>.</p>
</div>
<div class="section level2">
<h2 id="questions-and-feedback">Questions and Feedback<a class="anchor" aria-label="anchor" href="#questions-and-feedback"></a>
</h2>
<p>For questions or concerns with <code>callbackreproducibility</code> or the <code>callback</code> <code>R</code> package, please contact <a href="mailto:alan_denadel@brown.edu">Alan DenAdel</a> or <a href="lcrawford@microsoft.com">Lorin Crawford</a>. Any feedback on the manuscript or figure reproducibility is greatly appreciated.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing callbackreproducibility</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Alan DenAdel <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-7985-6789" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Alan DenAdel.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
